core/src/main/java/org/apache/iceberg/util/ContentFileUtil.java

```diff
@@ -49,7 +50,21 @@ public static <F extends ContentFile<K>, K> K copy(
     }
   }
 
+  /**
+   * @deprecated since 1.7.0, will be removed in 2.0.0; use {@link
+   *     #referencedDataFileLocation(DeleteFile)} instead.
+   */
+  @Deprecated
   public static CharSequence referencedDataFile(DeleteFile deleteFile) {
+    return referencedLocation(deleteFile);
+  }
+
+  public static String referencedDataFileLocation(DeleteFile deleteFile) {
+    CharBuffer location = referencedLocation(deleteFile);
+    return location != null ? location.toString() : null;
+  }
+
+  private static <T> T referencedLocation(DeleteFile deleteFile) {
```

Comment:
The logic in referencedLocation will change once we track referenced location per manifest entry, but that depends on the spec change. It does appear like for now there's a performance benefit to getting off the CharSequenceMap

Reply:
I think we can migrate now as well.